### PR TITLE
Move documentation generation to main repository

### PR DESCRIPTION
**Pull request by berquist**
This is the first step needed for https://github.com/cclib/cclib.github.io/issues/32.

This PR does two things:
1. Copies only the static files, the attributes/coverage file generators, and the Sphinx config from cclib.github.io, and
2. Updates the Makefile to be more like the one auto-generated by new versions of Sphinx. See https://github.com/cclib/cclib/commit/9db4c9873f531c26256fb9077d3cf05d7d20fc31.
3. Adds a script that builds the documentation and pushes it to the `gh-pages` branch of cclib.github.io. It shouldn't be necessary with Travis automation, but might come in useful.

### CODE REVIEW

**Comment by berquist** [comment]
This should be ready, since adding the scripts for Travis automation will make this PR too large.

**Comment by berquist** [comment]
When it's time, can you pick "rebase and merge" on this? I want to see what it looks like again and think about it...

File: doc/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Documentation for cclib
+
+This repository contains the source of the current official website and documentation for [cclib](https://github.com/cclib/cclib), available on GitHub pages at http://cclib.github.io.
```
Comment (langner) [inline doc/README.md]: reposiory->directory

**Comment by langner** [comment]
> When it's time, can you pick "rebase and merge" on this? I want to see what it looks like again and think about it...

**Comment by langner** [comment]
> When it's time, can you pick "rebase and merge" on this? I want to see what it looks like again and think about it...

Sure. I think it's ready now - let me know if there's something else we're waiting for.

**Comment by berquist** [comment]
> let me know if there's something else we're waiting for.

I didn't realize this. I think it's done from my side.

**Comment by berquist** [comment]
> Perhaps it makes sense to rebase in several commits in this case, if only to get rid of the ephemeral submodule in there.

I did an interactive rebase to remove them completely.

**Pull request merged**